src/components/ProductInfo/ProductInfo.component.jsx

```diff
@@ -0,0 +1,115 @@
+import { useState, useContext, useEffect } from "react";
+import CartContext from "../../state/CartContext";
+
+function ProductInfo(props) {
+  const { state, dispatch } = useContext(CartContext);
+  const [availableStock, setAvailableStock] = useState([1]);
+  const [selectedQuantity, setSelectedQuantity] = useState(1);
+
+  function handleSelect(e) {
+    setSelectedQuantity(parseInt(e.target.value));
+  }
+
+  function handleAddToCart() {
+    if (selectedQuantity > availableStock.length) {
+      alert("Not enough stock");
+      updateAvailableStock();
+    } else {
+      dispatch({
+        type: "ADD_TO_CART",
+        payload: {
+          id: props.id,
+          name: props.data.name,
+          image: props.data.mainimage.url,
+          price: props.data.price,
+          quantity: selectedQuantity,
+        },
+      });
+      updateAvailableStock();
+    }
```

Comment:
No matter what happens on the if condition, you are calling updateAvailableStock. Moving it outside the condition is better
```suggestion
    if (selectedQuantity > availableStock.length) {
      alert("Not enough stock");
    } else {
      dispatch({
        type: "ADD_TO_CART",
        payload: {
          id: props.id,
          name: props.data.name,
          image: props.data.mainimage.url,
          price: props.data.price,
          quantity: selectedQuantity,
        },
      });
    }
    updateAvailableStock();
```